experimental/attestation_common/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use openssl::{error::ErrorStack, nid::Nid, x509::X509Extension};
+use serde::{Deserialize, Serialize};
+use x509_parser::{
+    der_parser::{self, oid::Oid},
+};
+
+// Using [`Nid::NETSCAPE_COMMENT`] identifier since `rust-openssl` doesn't support custom
+// extensions yet.
+// https://github.com/sfackler/rust-openssl/issues/1411
+// https://www.alvestrand.no/objectid/2.16.840.1.113730.1.13.html
+pub const TEE_EXTENSION_OID: Oid<'static> = der_parser::oid!(2.16.840.1.113730.1.13);
+
+/// Placeholder implementation of TEE quote for remote attestation.
+#[derive(Deserialize, Serialize, Debug)]
+pub struct Quote {
+    pub measurement: String,
+    pub public_key_hash: String,
```

Comment:
Shouldn't these be arbitrary bytes? Why strings? I mean, it's fine to base64 or hex encode them when converting to JSON, but I don't think the struct should contain these pretty printed versions.

Reply:
Done